pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -61,32 +61,32 @@ func defaultPredicates() sets.String {
 // when this function is called, and should be called in tests which may modify the value
 // of a feature gate temporarily.
 func ApplyFeatureGates() (restore func()) {
-	snapshot := factory.Copy()
+	snapshot := scheduler.CopyRegisteredPredicatesAndPriorities()
```

Comment:
Could be in this PR or a follow up:

`scheduler.PredicatesAndPrioritiesSnapshot`, which returns the restore function already. I don't see any need to export `Snapshot`.

Reply:
Could be in a follow up PR as it's part of the next step 
>Unexport all functions and types that are only used from pkg/scheduler. There seem to be some unused functions already. So we can remove them here.

Reply:
sg